python/lsst/atmospec/nightAnalysis.py

```diff
@@ -78,56 +73,61 @@ def getLineValue(table, lineName, columnName, nanForMissingValues=True):
     return table[columnName][rowNum]
 
 
-class NightStellarSpectra():
+class NightStellarSpectra:
     """Class for holding the results for a night's observations of a given star
     """
 
-    def __init__(self, rerunDir, dayObs, targetName, *, butler=None, ignoreSeqNums=[]):
+    def __init__(self, rerunDir, dayObs, targetName, instrument, *, butler=None, ignoreSeqNums=[],
+                 collections=None):
         self.rerunDir = rerunDir
-        self.dayObs = dayObs
+        self.dayObs = int(dayObs)
         self.targetName = targetName
+        if instrument is None:
+            raise ValueError("An instrument name must be provided.")
+        self.instrument = instrument
+
         if butler:
-            self.butler = butler
+            self.butler = dafButler.Butler(butler=butler, instrument=instrument, collections=collections)
```

Comment:
Can you really reinstantiate a butler using a butler? That is interesting to know!